{"id":"20078261082","type":"IssuesEvent","actor":{"id":5675552,"login":"LeShadow","display_login":"LeShadow","gravatar_id":"","url":"https://api.github.com/users/LeShadow","avatar_url":"https://avatars.githubusercontent.com/u/5675552?"},"repo":{"id":66578799,"name":"open-quantum-safe/openssl","url":"https://api.github.com/repos/open-quantum-safe/openssl"},"payload":{"action":"opened","issue":{"url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351","repository_url":"https://api.github.com/repos/open-quantum-safe/openssl","labels_url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351/labels{/name}","comments_url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351/comments","events_url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351/events","html_url":"https://github.com/open-quantum-safe/openssl/issues/351","id":1124299120,"node_id":"I_kwDOA_fpb85DA3Fw","number":351,"title":"Integrating OpenSSL & Liboqs in Python - SSL Unexpected Message","user":{"login":"LeShadow","id":5675552,"node_id":"MDQ6VXNlcjU2NzU1NTI=","avatar_url":"https://avatars.githubusercontent.com/u/5675552?v=4","gravatar_id":"","url":"https://api.github.com/users/LeShadow","html_url":"https://github.com/LeShadow","followers_url":"https://api.github.com/users/LeShadow/followers","following_url":"https://api.github.com/users/LeShadow/following{/other_user}","gists_url":"https://api.github.com/users/LeShadow/gists{/gist_id}","starred_url":"https://api.github.com/users/LeShadow/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/LeShadow/subscriptions","organizations_url":"https://api.github.com/users/LeShadow/orgs","repos_url":"https://api.github.com/users/LeShadow/repos","events_url":"https://api.github.com/users/LeShadow/events{/privacy}","received_events_url":"https://api.github.com/users/LeShadow/received_events","type":"User","site_admin":false},"labels":[],"state":"open","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":0,"created_at":"2022-02-04T15:01:53Z","updated_at":"2022-02-04T15:01:53Z","closed_at":null,"author_association":"NONE","active_lock_reason":null,"body":"Hey!\r\n\r\nSo for the last few days I have been trying to integrate liboqs and this version of OpenSSL in Python 3.8.11. \r\nThis is basically what I am doing:\r\n\r\n- Configure and compile liboqs with the following configure settings: `cmake -GNinja -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=../../openssl/oqs ..`\r\n- Configure and compile openssl with the following configure settings: `./Configure shared linux-x86_64 -lm --prefix=/opt/openssl_installed --openssldir=/opt/openssl_installed`\r\n\r\nNow when I compile Python, it finds that custom openssl installation, I added the paths to my PATH as well.\r\n\r\nHowever, when I then use Python, every SSL connection (even the ones without PQC) are getting an error with SSL \"UNEXPECTED MESSAGE\".\r\n\r\nExample: `Starting new HTTPS connection (1): pypi.org:443\r\nIncremented Retry for (url='/simple/requests/'): Retry(total=4, connect=None, read=None, redirect=None, status=None)\r\nWARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: UNEXPECTED_MESSAGE] unexpected message (_ssl.c:1131)'))': /simple/requests/`\r\n\r\nI was wondering if anyone could give me pointers, or get me on the right path, on how to fix this?\r\n\r\nThank you!","reactions":{"url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351/reactions","total_count":0,"+1":0,"-1":0,"laugh":0,"hooray":0,"confused":0,"heart":0,"rocket":0,"eyes":0},"timeline_url":"https://api.github.com/repos/open-quantum-safe/openssl/issues/351/timeline","performed_via_github_app":null}},"public":true,"created_at":"2022-02-04T15:01:53Z","org":{"id":20689385,"login":"open-quantum-safe","gravatar_id":"","url":"https://api.github.com/orgs/open-quantum-safe","avatar_url":"https://avatars.githubusercontent.com/u/20689385?"}}
